subsys/shell/modules/device_service.c

```diff
@@ -23,6 +23,16 @@ extern const struct device __device_end[];
 extern const struct device __device_SMP_start[];
 #endif
 
+enum init_level {
+	INIT_LEVEL_PRE_KERNEL_1,
+	INIT_LEVEL_PRE_KERNEL_2,
+	INIT_LEVEL_POST_KERNEL,
+	INIT_LEVEL_APPLICATION,
+#ifdef CONFIG_SMP
+	INIT_LEVEL_SMP,
+#endif
+};
```

Comment:
I don't see why enum init_level cannot be public, here for instance it misses the EARLY and makes this confusing.

Reply:
Precisely because it misses EARLY the code actually works now (`main` is currently broken). These are indices of an array internal to `z_sys_init_run_level`, not public anymore after these changes. The device service shell creates its own array, and as explained in the commit, it's up to the module on how to handle access to such array. They should only be public if used as identifiers, not tied to any implementation indexing technique.